pkg/library/flatten/merge.go

```diff
@@ -165,7 +165,8 @@ func needsContainerContributionMerge(flattenedSpec *dw.DevWorkspaceTemplateSpec)
 				return false, fmt.Errorf("failed to parse %s attribute on component %s as true or false", constants.MergeContributionAttribute, component.Name)
 			}
 		} else {
-			if !component.Attributes.Exists(constants.PluginSourceAttribute) {
+			pluginSource := component.Attributes.GetString(constants.PluginSourceAttribute, nil)
+			if pluginSource == "" || pluginSource == "parent" {
```

Comment:
It doesn't really sit well to me that the merge target found by `needsContainerContributionMerge()` and  `findMergeTarget` may be different, but I'm not sure if that affects things. Regardless, I'm going to fix this disparity shortly